packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java

```diff
@@ -140,6 +142,101 @@ public static class NotificationProperties {
 		private UUID nodeId;
 	}
 
+	@Data
+	static class EquationRef {
+
+		private String id;
+		private String equationStr;
+	}
+
+	@Data
+	static class EquationToModelInput {
+
+		private UUID nodeId;
+		private UUID workflowId;
+		private Map<UUID, List<EquationRef>> equationsWithSource;
+		private List<String> equations;
+	}
+
+	@PostMapping("/equations-to-model-new")
+	@Secured(Roles.USER)
+	@HasProjectAccess(level = Schema.Permission.WRITE)
+	public ResponseEntity<UUID> equationsToModelNew(
```

Comment:
May be worth adding a small doc string .. Though I dont think many of our java classes have them so up to you